libtransact/src/contract/archive/mod.rs

```diff
@@ -14,3 +14,586 @@
+    /// and no pre-release tag.
+    #[test]
+    fn find_scar_insufficient_release() {
+        let dir = new_temp_dir();
+        write_mock_scar(&dir, "mock", "0.1.0-dev");
+
+        assert!(find_scar("mock", "0.1.0", &dir).is_err());
+    }
+
+    /// Verify that an error is returned when no contract is found with a sufficient patch version.
+    #[test]
+    fn find_scar_insufficient_patch() {
+        let dir = new_temp_dir();
+        write_mock_scar(&dir, "mock", "0.1.0");
+
+        assert!(find_scar("mock", "0.1.1", &dir).is_err());
+    }
+
+    /// Verify that an error is returned when no contract is found with a sufficient minor version.
+    #[test]
+    fn find_scar_insufficient_minor() {
+        let dir = new_temp_dir();
+        write_mock_scar(&dir, "mock", "0.1.0");
+
+        assert!(find_scar("mock", "0.2.0", &dir).is_err());
+    }
+
+    /// Verify that an error is returned when no contract is found with a sufficient major version.
+    #[test]
+    fn find_scar_insufficient_major() {
+        let dir = new_temp_dir();
+        write_mock_scar(&dir, "mock", "0.1.0");
+
+        assert!(find_scar("mock", "1.0.0", &dir).is_err());
+    }
+
+    /// Verify that a .scar file is correctly found when any version (`*`) is requested.
+    #[test]
+    fn find_scar_any_version() {
+        let dir = new_temp_dir();
+        let scar_path = write_mock_scar(&dir, "mock", "0.1.0");
+
+        assert_eq!(
+            find_scar("mock", "*", &dir).expect("failed to find .scar"),
+            scar_path
+        );
+    }
+
+    /// Verify that a .scar file is correctly found when the exact version is requested.
+    #[test]
+    fn find_scar_exact_version() {
+        let dir = new_temp_dir();
+        let scar_path = write_mock_scar(&dir, "mock", "0.1.2");
+
+        assert_eq!(
+            find_scar("mock", "0.1.2", &dir).expect("failed to find .scar"),
+            scar_path
+        );
+    }
+
+    /// Verify that a .scar file is correctly found when it meets the minimum minor version.
+    #[test]
+    fn find_scar_minimum_minor() {
+        let dir = new_temp_dir();
+        let scar_path = write_mock_scar(&dir, "mock", "0.1.2");
+
+        assert_eq!(
+            find_scar("mock", "0.1", &dir).expect("failed to find .scar"),
+            scar_path
+        );
+    }
+
+    /// Verify that a .scar file is correctly found when it meets the minimum major version.
+    #[test]
+    fn find_scar_minimum_major() {
+        let dir = new_temp_dir();
+        let scar_path = write_mock_scar(&dir, "mock", "1.2.3");
+
+        assert_eq!(
+            find_scar("mock", "1", &dir).expect("failed to find .scar"),
+            scar_path
+        );
+    }
+
+    /// Verify that the .scar file with no pre-release tag is returned rather than the same patch
+    /// version with a pre-release tag.
+    #[test]
+    fn find_scar_highest_matching_patch() {
+        let dir = new_temp_dir();
+        write_mock_scar(&dir, "mock", "0.1.2-dev");
```

Comment:
Can you add some tests where the -dev version would be picked?

Reply:
Added (`find_scar_pre_release`)